tests/test_memcache.py

```diff
@@ -58,6 +59,34 @@ def test_delete(self):
         self.assertEqual(result, True)
         self.assertEqual(self.mc.get("long"), None)
 
+        # Delete with explicit time=0 (can re-set immediately)
+        self.mc.set("my_key", "my_val")
+        self.mc.delete("my_key", time=0)
+        self.assertNotEqual(self.mc.set("my_key", "my_val"), 0)
+
+    def test_touch(self):
+        # Basic operation
+        self.mc.set("my_key", "my_val", time=1)
+        self.mc.touch("my_key", time=3)
+        time.sleep(2)
+        self.assertEqual(self.mc.get("my_key"), "my_val")  # It's been prolonged...
+        time.sleep(2)
+        self.assertEqual(self.mc.get("my_key"), None)  # But finally expires
+
+        # Default time
+        self.mc.set("my_key2", "my_val", time=1)
+        self.mc.touch("my_key2")
+        time.sleep(2)
+        self.assertEqual(self.mc.get("my_key2"), "my_val")
+        # TODO: test it stays forever
```

Comment:
I thought using mocking was larger than this issue and could be solved in another PR, but I can give it a try here!